### ISSUE TITLE
Invalid "argument.type.incompatible" - thinks it can be null

### ISSUE THREAD

**Issue by scottkennedy** [typetools/checker-framework#736]
Test case running on 2.0.0:

```java
public class Test {
    private static void doSomething(@NonNull Object[] objects) {
        // ...
    }

    public static void something(@Nullable Object[] objects) {
        if (objects != null) {
            doSomething(objects);
        }
    }
}
```

Error:

```
Test.java:13: error: [argument.type.incompatible] incompatible types in argument.
            doSomething(objects);
                        ^
  found   : @Initialized @Nullable Object @Initialized @NonNull []
  required: @Initialized @NonNull Object @Initialized @NonNull []
```

**Comment by smillst** [comment]
This error is expected.  `@Nullable Object[]`, which is defaulted to `@Nullable Object @NonNull []`, is a non-null array of possibly null elements.  `objects != null` does not test whether any of the elements of `objects` are non-null.

This code below type checks:
````
public class Test {
    private static void doSomething(@Nullable Object[] objects) {
        // ...
    }

    public static void something(@Nullable Object @Nullable[] objects) {
        if (objects != null) {
            doSomething(objects);
        }
    }
}
````
